File: MyApplication/app/src/main/java/devandroid/rafael/myapplication/view/MainActivity.java
package devandroid.rafael.myapplication.view;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import devandroid.rafael.myapplication.R;
import devandroid.rafael.myapplication.model.Pessoa;

public class MainActivity extends AppCompatActivity {


    Pessoa pessoa;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);





        pessoa = new Pessoa();


        editPrimeroNome = findViewById(R.id.editPrimeroNome);
        editSobrenome = findViewById(R.id.editSobrenome);
        editNomeCurso = findViewById(R.id.editNomeCurso);
        editTelefone = findViewById(R.id.editTelefone);


        btnLimpar = findViewById(R.id.btnLimpar);
        btnSalvar = findViewById(R.id.btnSalvar);
        btnFinalizar = findViewById(R.id.btnFinalizar);


        btnLimpar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                editPrimeroNome.setText("");
                editSobrenome.setText("");
                editTelefone.setText("");
                editNomeCurso.setText("");
            }
        });



        btnSalvar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                pessoa.setPrimeiroNome(editPrimeroNome.getText().toString());
                pessoa.setSobreNome(editSobrenome.getText().toString());
                pessoa.setCursoDesejado(editNomeCurso.getText().toString());
                pessoa.setTelefoneContato(editTelefone.getText().toString());

                Toast.makeText(MainActivity.this, "Salvo", Toast.LENGTH_SHORT).show();
            }
        });


        btnFinalizar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                Toast.makeText(MainActivity.this, "Volte Sempre ", Toast.LENGTH_SHORT).show();
                finish();

            }
        });



    }

    EditText editPrimeroNome;
    EditText editSobrenome;
    EditText editNomeCurso;
    EditText editTelefone;

    Button btnSalvar;
    Button btnLimpar;
    Button btnFinalizar;

}

File: RickAndMorty/app/src/main/java/com/example/rickandmorty22101195/MainActivity.kt
package com.example.rickandmorty22101195

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import com.example.rickandmorty22101195.dadosremotos.Personagem
import com.example.rickandmorty22101195.dadosremotos.retrofitService
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class MainActivity : AppCompatActivity() {

    lateinit var editTextPersonagemID: EditText
    lateinit var buttonBuscar: Button
    lateinit var textViewNomeValor: TextView
    lateinit var textViewStatusValor: TextView
    lateinit var textViewEspeciesValor: TextView
    lateinit var textViewGeneroValor: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        editTextPersonagemID = findViewById(R.id.editTextPersonagemID)
        buttonBuscar = findViewById(R.id.buttonBuscar)
        textViewNomeValor = findViewById(R.id.textViewNomeValor)
        textViewStatusValor = findViewById(R.id.textViewStatusValor)
        textViewEspeciesValor = findViewById(R.id.textViewEspeciesValor)
        textViewGeneroValor = findViewById(R.id.textViewGeneroValor)

        buttonBuscar.setOnClickListener{
            var idDigitado: String = editTextPersonagemID.text.toString()

            retrofitService.getPersonagem(idDigitado.toInt()).enqueue(object : retrofit2.Callback<Personagem> {
                override fun onResponse(call: Call<Personagem>, response: Response<Personagem>) {
                    if(response.isSuccessful) {
                        textViewNomeValor.text = response.body()!!.nome
                        textViewStatusValor.text = response.body()!!.vivo
                        textViewEspeciesValor.text = response.body()!!.especie
                        textViewGeneroValor.text = response.body()!!.genero


                    } else{
                        Toast.makeText(this@MainActivity, "Ocorreu um erro", Toast.LENGTH_SHORT).show()
                    }
                }

                override fun onFailure(call: Call<Personagem>, t: Throwable) {
                    Toast.makeText(this@MainActivity, "Sem internet", Toast.LENGTH_SHORT).show()
                }

            })
        }



    }
}